internal/controller/storageclient_controller.go

```diff
@@ -369,43 +393,43 @@ func (r *storageClientReconcile) reconcilePhases() (ctrl.Result, error) {
 
 	r.storageClient.Status.InMaintenanceMode = storageClientResponse.MaintenanceMode
 
-	for _, kubeResource := range storageClientResponse.KubeResources {
-		var err error
-		typeMeta := &metav1.TypeMeta{}
-		if err := json.Unmarshal(kubeResource, typeMeta); err != nil {
-			return reconcile.Result{}, fmt.Errorf("failed to unmarshal the type for the Object: %w", err)
-		}
-
-		// Create the received resources, if necessary.
-		switch typeMeta.GroupVersionKind() {
-		case quotav1.SchemeGroupVersion.WithKind("ClusterResourceQuota"):
-			err = r.reconcileResource(&quotav1.ClusterResourceQuota{}, kubeResource)
-		case csiopv1a1.GroupVersion.WithKind("CephConnection"):
-			err = r.reconcileResource(&csiopv1a1.CephConnection{}, kubeResource)
-		case csiopv1a1.GroupVersion.WithKind("ClientProfileMapping"):
-			err = r.reconcileResource(&csiopv1a1.ClientProfileMapping{}, kubeResource)
-		case corev1.SchemeGroupVersion.WithKind("Secret"):
-			err = r.reconcileResource(&corev1.Secret{}, kubeResource)
-		case nbv1.SchemeGroupVersion.WithKind("Noobaa"):
-			err = r.reconcileResource(&nbv1.NooBaa{}, kubeResource)
-		case storagev1.SchemeGroupVersion.WithKind("StorageClass"):
-			err = r.reconcileResource(&storagev1.StorageClass{}, kubeResource)
-		case snapapi.SchemeGroupVersion.WithKind("VolumeSnapshotClass"):
-			err = r.reconcileResource(&snapapi.VolumeSnapshotClass{}, kubeResource)
-		case groupsnapapi.SchemeGroupVersion.WithKind("VolumeGroupSnapshotClass"):
-			if val, _ := r.crdsBeingWatched.Load(VolumeGroupSnapshotClassCrdName); !val.(bool) {
+	requestsPerGvk := map[string][]reconcileRequest{}
+	for _, kubeObj := range storageClientResponse.KubeObjects {
+		if kubeObj != nil {
+			objectMeta := &metav1.PartialObjectMetadata{}
+			if err := json.Unmarshal(kubeObj.Bytes, objectMeta); err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to unmarshal metadata for the Object: %w", err)
+			}
+			gvk := objectMeta.GroupVersionKind().String()
+			requestsPerGvk[gvk] = append(
+				requestsPerGvk[gvk],
+				reconcileRequest{
+					name:      objectMeta.GetName(),
+					namespace: objectMeta.GetNamespace(),
+					bytes:     kubeObj.Bytes,
+				},
+			)
+		}
+	}
+	var combinedErr error
+	for idx := range kindsToReconcile {
+		kind := &kindsToReconcile[idx]
+		if kind.requiresCrd != "" {
+			if val, _ := r.crdsBeingWatched.Load(kind.requiresCrd); !val.(bool) {
```

Comment:
Can't we handle that by analyzing the get return error instead?
I am asking because the rest mapper should be the source of truth

Reply:
it is stored in this syncmap only after doing a get on the clustter